content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-flows.md

```diff
@@ -449,6 +449,181 @@ data class CreateBoardingTicketInitiator @JsonConstructor constructor(private va
 }
 ```
 
+## Write the `GiftVoucherToFriend` flow
+
+In Corda, you can transfer a state to another party on the network. The `GiftVoucherToFriend` flow lets you transfer the ownership of the issued voucher to another person.
+
+Now that you've written the `CreateAndIssueMarsVoucher` and `CreateBoardingTicket` flows, try writing the `GiftVoucherToFriend` flow.
+
+You will need these variables:
+
+* `voucherID`: ID of the issued voucher whose ownership is to be transferred.
+* `holder`: The new voucher owner.
+
+You must inject these services:
+
+* `FlowEngine`: Provides basic control over a flow it is injected into, such as calling subflows, performing asynchronous tasks and putting the flow to sleep for a period of time. This flow needs `FlowEngine` because you are utilizing subflows such as `FinalityFlow`.
+* `FlowIdentity`: Obtains the identity of the node running the flow. You must inject this service because one of the output states needs its own identity.
+* `FlowMessaging`: Used for creating and closing flow sessions as well as sending and receiving data between flows via a flow session. Use this service to create flow sessions that allow communication between counterparties.
+* `TransactionBuilderFactory`: Used for constructing, verifying, and signing transactions. In this case, it builds a transaction.
+* `IdentityService`: Provides methods to retrieve Party and AnonymousParty instances. You need it because your flow involves counterparties.
```

Comment:
Should Party and AnonymousParty be shown as code?

Reply:
good call. yes, it should.